### DISCUSSION TITLE
Setting Initial Condition on Unreal Engine

### DISCUSSION THREAD

**Discussion by nmfarras** [JSBSim-Team/jsbsim#1057] [Q&A] [answered]
I want to setting up my aircraft at initial altitude and calibrated speed just like one from reset file. I try to implement the initial condition like this

implementation on init:
![image](https://github.com/JSBSim-Team/jsbsim/assets/30061607/3f2af958-eab8-4c76-b44c-5725e3eaa8aa)

The detail of `Initialization JSB`:
![image](https://github.com/JSBSim-Team/jsbsim/assets/30061607/013b2743-133e-40d2-a737-92a0b3b76dcd)
 
Unfortunately end up with following error
![image](https://github.com/JSBSim-Team/jsbsim/assets/30061607/60d0907d-5660-4951-87da-bf7236361de2)
 
The app run normally again when i remove this node. Any suggestion how to correctly set this? I use command console batch block for this.

Thanks in advance!

**Comment by gallonmate** [comment, marked as answer]
UE5 JSBSim plugin has the initialization stage written in c++. I'm not sure you can currently make it work with that method of nodes.
But anyways when the Command Console node crashes, it's because the property doesn't exist or it's read only. In this case it look like there is an extra blank space in front of the first string of `ic/h`, which will crash the engine, because "_ic/h" doesn't exist.

That being said, the plugin already has a variable to set the initial calibrated airspeed in the JSBSim component. You can set it in the blueprint or in the level editor details if the blueprint is placed in the level. As for setting initial altitude, you can set the Z height in UE5 for the blueprint. 1 unit equals 1 cm. So 100000 units in Z equal 1km in altitude.

**Reply by nmfarras** [reply]
that's really silly of me. I just realize the extra space. But after I fix it like following
![image](https://github.com/JSBSim-Team/jsbsim/assets/30061607/904bda41-8cc3-4a34-b4ef-6c3efc312b04)

the error still exactly the same as the previous one. Does that mean `ic/h-sl-ft` not valid? i thought it is one since from catalogue shows RW
![image](https://github.com/JSBSim-Team/jsbsim/assets/30061607/61b1c8c5-b8c9-42f8-b14c-80ca9375b284)

But anyways I try your recommendation and its work! Thank  you!

**Reply by gallonmate** [reply]
It probably crashes because the property isn't loaded yet. JSBSim has to load first. And I found getting/setting properties through command console too early can cause crash.

**Comment by nmfarras** [comment]
@gallonmate I want to ask for the following part
![image](https://github.com/JSBSim-Team/jsbsim/assets/30061607/fbc03a6e-62fd-4e61-9791-9171959e1cb0)

Where can I change the command part? Idk why but I cant change it. The value keep changing back to the initial value.

**Reply by gallonmate** [reply]
Yes it's some problem/unexpected behavior with this section. These commands you can set them in the main blueprint instead of the level editor.

**Reply by nmfarras** [reply]
You are right. Thank you again!
